awx/main/models/credential/__init__.py

```diff
@@ -33,11 +34,34 @@
     ROLE_SINGLETON_SYSTEM_AUDITOR,
 )
 from awx.main.utils import encrypt_field
+from . import injectors as builtin_injectors
 
-__all__ = ['Credential', 'CredentialType', 'V1Credential']
+__all__ = ['Credential', 'CredentialType', 'V1Credential', 'build_safe_env']
 
 logger = logging.getLogger('awx.main.models.credential')
 
+HIDDEN_PASSWORD = '**********'
+
+
+def build_safe_env(env):
+    '''
+    Build environment dictionary, hiding potentially sensitive information
+    such as passwords or keys.
+    '''
+    hidden_re = re.compile(r'API|TOKEN|KEY|SECRET|PASS', re.I)
```

Comment:
So if a user defines a custom credential and injects env vars that looks like any of these, i.e. `MY_TOKEN`, `PASSDOWN`, `KEYSTONE` it will get hidden. I guess that's how it has always worked.

Reply:
Yea, that's one difference here.  _Currently_ we only filter env vars for custom credentials that explicitly set `secret` on the field definition.  But honestly, I think it's fine to err on the side of caution here.